src/dmd/parse.d

```diff
@@ -9010,6 +9012,31 @@ final class Parser(AST) : Lexer
             token.lineComment = null;
         }
     }
+
+    import core.stdc.stdarg;
+
+    package final void error(string format, ...)
+    {
+        va_list args;
+        va_start(args, format);
+        lexer.diagnosticReporter.error(lexer.token.loc, format.ptr, args);
+        va_end(args);
+    }
+
+    package final void error(const ref Loc loc, string format, ...)
+    {
+        va_list args;
+        va_start(args, format);
+        lexer.diagnosticReporter.error(loc, format.ptr, args);
+        va_end(args);
+    }
+    private final void deprecation(const(char)* format, ...)
+    {
+        va_list args;
+        va_start(args, format);
+        lexer.diagnosticReporter.deprecation(lexer.token.loc, format, args);
+        va_end(args);
+    }
```

Comment:
Are these three methods necessary? I would think the methods in the lexer would be called automatically due to the `alias this`.

Reply:
I had hoped that but the seem to pick up `dmd.errors.error` & friends.

Reply:
Note that 
``` 
test.d(5):        There's no `static else`, use `else` instead.
```
 is a supplemental error, which is not defined above.